stan/math/gpu/opencl_context.hpp

```diff
@@ -73,13 +73,21 @@ class opencl_context_base {
     try {
       // platform
       cl::Platform::get(&platforms_);
+      if (OPENCL_PLATFORM_ID >= platforms_.size()) {
+        system_error("OpenCL Initialization", "[Platform]", -1,
+                       "CL_INVALID_PLATFORM");
+      }
       platform_ = platforms_[OPENCL_PLATFORM_ID];
       platform_name_ = platform_.getInfo<CL_PLATFORM_NAME>();
       platform_.getDevices(DEVICE_FILTER, &devices_);
       if (devices_.size() == 0) {
         system_error("OpenCL Initialization", "[Device]", -1,
                      "CL_DEVICE_NOT_FOUND");
       }
+      if (OPENCL_DEVICE_ID >= devices_.size()) {
+        system_error("OpenCL Initialization", "[Device]", -1,
+                       "CL_INVALID_DEVICE");
+      }
```

Comment:
If a user specified an ID that did not exist, the test produced a seg fault. We now output the following:

[ RUN      ] MathGpu.getInfo
unknown file: Failure
C++ exception with description "opencl_context: clGetDeviceIDs CL_DEVICE_NOT_FOUND: Unknown error -1" thrown in the test body.
[  FAILED  ] MathGpu.getInfo (31 ms)